### ISSUE TITLE
[BUG] Inliner fails to inline operators

### ISSUE THREAD

**Issue by konnov** [informalsystems/apalache#283]
Somewhat unexpected, the following simple piece of code does not work:

```tla
----------------------------- MODULE Callback ----------------------------      
EXTENDS Integers

VARIABLES x

Pick(Cb(_)) ==
    \E i \in 1..10:
        Cb(i)

Init ==
    LET Cb(i) ==
        x = i
    IN
    Pick(Cb)

Next ==
    LET Cb(i) ==
        x' = i
    IN
    Pick(Cb)

Inv ==
    x > 0
==========================================================================
```

**Comment by Kukovec** [comment]
While the specification seems to violate `FlatLanguagePredicate`, it is not an issue with inlining, as the inliner produces the expected output.
We should consider removing (higher-order) operators  from the module, once they have been inline in `Init` and `Next`.

**Comment by konnov** [comment]
Thanks! We should filter out the operators that are not required for the further model checking pass. I will pick this issue.

**Comment by konnov** [comment]
This is fixed in #301
